glean-core/src/upload/mod.rs

```diff
@@ -140,6 +145,100 @@ impl PingUploadManager {
             None => PingUploadTask::Done,
         }
     }
+
+    /// Processes the response from an attempt to upload a ping.
+    ///
+    /// Spawns a thread for any disk IO necessary.
+    ///
+    /// Based on the HTTP status of said response,
+    /// the possible outcomes are:
+    ///
+    /// * **200 - 299 Success**
+    ///   Any status on the 2XX range is considered a succesful upload,
+    ///   which means the corresponding ping file can be deleted.
+    ///   _Known 2XX status:_
+    ///   * 200 - OK. Request accepted into the pipeline.
+    ///
+    /// * **400 - 499 Unrecoverable error**
+    ///   Any status on the 4XX range means something our client did is not correct.
+    ///   It is unlikely that the client is going to recover from this by retrying,
+    ///   so in this case the corresponding ping file can also be deleted.
+    ///   _Known 4XX status:_
+    ///   * 404 - not found - POST/PUT to an unknown namespace
+    ///   * 405 - wrong request type (anything other than POST/PUT)
+    ///   * 411 - missing content-length header
+    ///   * 413 - request body too large Note that if we have badly-behaved clients that
+    ///           retry on 4XX, we should send back 202 on body/path too long).
+    ///   * 414 - request path too long (See above)
+    ///
+    /// * **Any other error**
+    ///   For any other error, a warning is logged and the ping is re-enqueued.
+    ///   _Known other errors:_
+    ///   * 500 - internal error
+    ///
+    /// # Arguments
+    ///
+    /// `uuid` - The UUID of the ping in question.
+    /// `status` - The HTTP status of the response.
+    ///
+    /// # Return value
+    ///
+    /// Returns the JoinHandle for the spawned thread for test syncronization purposes.
+    ///
+    /// # Panics
+    ///
+    /// Will panic if unable to spawn a new thread.
+    pub fn process_ping_upload_response(&self, uuid: &str, status: u16) -> thread::JoinHandle<()> {
+        let mut task = "";
```

Comment:
What would be a better way to implement this? With this approach I get a lint error about unused_assignments.

Reply:
Don't use a string for what is essentially an enum to begin with.